src/consent-manager-builder/analytics.ts

```diff
@@ -32,6 +34,13 @@ export default function conditionallyLoadAnalytics({
   }
 
   for (const destination of destinations) {
+    // Was a preference explicitly set on this destination?
```

Comment:
I wonder if this logic around defaultDestinationBehavior could be moved into src/consent-manager-builder/index.tsx? That way most of the logic & responsibility for handling this is within a single file/module?